packages/knip/src/plugins/eleventy/index.ts

```diff
@@ -32,19 +33,39 @@ const findEleventyDependencies: GenericPluginCallback = async (configFilePath, o
     return config.entry
       ? config.entry.map(toProductionEntryPattern)
       : PRODUCTION_ENTRY_FILE_PATTERNS.map(toProductionEntryPattern);
-  if (typeof localConfig === 'function') localConfig = await localConfig(new DummyEleventyConfig());
+
+  const dummyUserConfig = new DummyEleventyConfig();
+  if (typeof localConfig === 'function') localConfig = await localConfig(dummyUserConfig);
 
   const inputDir = localConfig?.dir?.input || defaultEleventyConfig.dir.input;
   const dataDir = localConfig?.dir?.data || defaultEleventyConfig.dir.data;
   const templateFormats = localConfig.templateFormats || defaultEleventyConfig.templateFormats;
 
-  return (
-    config?.entry ?? [
-      join(inputDir, dataDir, '**/*.js'),
-      join(inputDir, `**/*.{${typeof templateFormats === 'string' ? templateFormats : templateFormats.join(',')}}`),
-      join(inputDir, '**/*.11tydata.js'),
-    ]
-  ).map(toEntryPattern);
+  const copiedPackages = [];
+  const copiedEntries = [];
+
+  for (const path of Object.keys(dummyUserConfig.passthroughCopies).map(normalize)) {
+    const parts = path.split('/');
+    if (path.startsWith('node_modules/') && parts.length > 1) {
+      copiedPackages.push(parts.at(1) as string);
+    } else {
+      if (path.endsWith('/')) {
+        copiedEntries.push(`${path}*.{${DEFAULT_EXTENSIONS.join(',')}}`);
+      } else copiedEntries.push(path);
```

Comment:
If it isn't in node_modules but it is a directory, add a glob that looks for all files with one of the DEFAULT_EXTENSIONS within that directory. If it isn't in node_modules or a directory, add it to our list of entry files.